### ISSUE TITLE
[plugin add] Extract binary from a .tar.gz / .zip

### ISSUE THREAD

**Issue by tcarreira** [tsuru/tsuru-client#173]
Some tsuru-plugins (eg: [rpaasv2](https://github.com/tsuru/rpaas-operator/releases/)) have the plugin binary compiled for multiple platforms, and they pack every binary in a tar.gz. That way, the final binary have the same name (`rpaasv2`) despite the fact there are multiple different binaries.

The tsuru client could support unpack this tar.gz and extract the `plugin_name` binary from inside.

---
proposed solution

```
download from <URL>
if file is tar.gz:
  unpack file to tempDir
  install tempDir/<plugin_name>
else
  install file
```

---
edit: in order to continue to provide multi-platform compatibility (aka: windows) we must support unpack from a .zip as well

**Comment by otherpirate** [comment]
@tcarreira I had 2 doubts about it
1) What if the tar.gz has directories? should I skip it? return error?
1) What if the tar.gz has more than one file? (eg readme.txt) should I return an error? skip?

**Comment by tcarreira** [comment]
closed by #191
